### PR TITLE
Support HAVE_SESSION_TICKET without realloc

### LINKED ISSUE #6209: Support `HAVE_SESSION_TICKET` without realloc
### Version

master

### Description

We should support session tickets even if the platform has no realloc: XREALLOC should be guarded and implemented by XMALLOC equivalent code if XREALLOC is not defined.

### PR DESCRIPTION
**Pull request by embhorn**
# Description

Supports session tickets even if the platform has no realloc: `XREALLOC` is guarded and implemented by `XMALLOC` code if `XREALLOC` is not defined.

Fixes #6209 

# Testing

```
./configure --enable-session-ticket CFLAGS="-DWOLFSSL_NO_REALLOC"
```

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by embhorn** [comment]
Retest this please

**Review by rizlik — APPROVED** [review]

File: src/ssl.c [hunk 1]
```diff
@@ -21497,11 +21497,23 @@ static int wolfSSL_DupSessionEx(const WOLFSSL_SESSION* input,
                 ret = WOLFSSL_FAILURE;
             }
             else {
+#ifndef XREALLOC
+                tmp = (byte*)XMALLOC(input->ticketLen,
+                        output->heap, DYNAMIC_TYPE_SESSION_TICK);
+                if (tmp != NULL)
+                    XMEMCPY(tmp, ticBuff, input->ticketLen);
```
Comment (JacobBarthelmeh) [inline src/ssl.c:21504]: Should this copy ticLenAlloc instead of input->ticketLen to try and avoid reading over the end of 'ticBuff'?

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 1 above)
Comment (embhorn) [reply src/ssl.c:21504]: Good catch! Updated and also fixed a build issue with `WOLFSSL_NO_REALLOC` defined.

Comment (rizlik) [reply src/ssl.c:21504]: yeah, good catch I missed it during my review.  Is this copy really necessary then? (is it overwritten by the memcpy on 21529? )

Comment (embhorn) [reply src/ssl.c:21504]: Hmmm. Interesting. Yeah, I hadn't even considered that. I love removing memcpy's

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**